rest_framework/views.py

```diff
@@ -511,6 +513,44 @@ def dispatch(self, request, *args, **kwargs):
         self.response = self.finalize_response(request, response, *args, **kwargs)
         return self.response
 
+    async def async_dispatch(self, request, *args, **kwargs):
+        """
+        `.dispatch()` is pretty much the same as Django's regular dispatch,
+        but with extra hooks for startup, finalize, and exception handling.
+        """
+        self.args = args
+        self.kwargs = kwargs
+        request = self.initialize_request(request, *args, **kwargs)
+        self.request = request
+        self.headers = self.default_response_headers  # deprecate?
+
+        try:
+            self.initial(request, *args, **kwargs)
+
+            # Get the appropriate handler method
+            if request.method.lower() in self.http_method_names:
+                handler = getattr(self, request.method.lower(),
+                                  self.http_method_not_allowed)
+            else:
+                handler = self.http_method_not_allowed
+
+            if asyncio.iscoroutinefunction(handler):
+                response = await handler(request, *args, **kwargs)
+            else:
+                raise Exception('Async methods should be used on an async view.')
```

Comment:
Due to the checks performed within Django core's `view_is_async`, this check _probably_ isn't needed. Unless you've found some edge case that I'm not aware of.

Reply:
I think it is redundant as well, so I will remove. Thanks.